src/egain/webChat/client/src/egainServiceDesk.ts

```diff
@@ -0,0 +1,392 @@
+      }
+      return true;
+    } catch (error) {
+      console.error('error: ', error);
+      return Promise.reject();
+    }
+  }
+
+  async handleMessage(event: string): Promise<void> {
+    if (event.includes('href=')) {
+      const link = this.getLink(event);
+      this.callback.sendMessageToUser(stringToMessageResponseFormat(link), '');
+    } else {
+      let message = this.sanitize(JSON.stringify(event));
+      try {
+        message = JSON.parse(message);
+      } catch (error) {
+        console.error('An error occurred while parsing the message: ', error);
+        return Promise.reject(error);
+      }
+
+      let messageContent;
+      let messageType;
+
+      if (this.sessionID === message.authorization) {
+        if ('status' in message) {
+          this.callback.agentTyping(false);
+          if (message.status === 'Conversation ended') {
+            this.socket.close();
+          }
+        } else if ('eGainMessage' in message) {
+          messageType = message.eGainMessage.type;
+          if (messageType === 'agent.unavailable') {
+            messageContent = message.eGainMessage.value.reason;
+            this.areAnyAgentsOnline(false);
+          } else if (messageType === 'conversation.state') {
+            messageContent = message.eGainMessage.value.status;
+            if (messageContent === 'escalated') {
+              this.callback.agentJoined(this.agentProfile);
+            }
+          } else if (messageType === 'text') {
+            this.callback.agentTyping(false);
+            this.agentProfile.nickname = message.eGainMessage.agentName;
+            messageContent = message.eGainMessage.value;
+            if (messageContent === 'agent.end.conversation') {
+              this.agentEnded = true;
+              this.callback.agentEndedChat();
+            } else {
+              this.callback.sendMessageToUser(stringToMessageResponseFormat(messageContent), '');
+            }
+          } else if (messageType === 'typing.start') {
+            this.callback.agentTyping(true);
+          } else if (messageType === 'typing.end') {
+            this.callback.agentTyping(false);
+          } else {
+            console.error('Unknown message type');
+          }
+        }
+      } else {
+        return Promise.reject();
+      }
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   * This method is used to parse hyperlinks
+   *
+   * @returns Returns a string.
+   */
+  getLink(link: string): string {
+    const root = parse(link);
+    const result = root.childNodes[1].rawText;
+    return result;
+  }
+
+  sanitize(message: string): string {
+    return message.replace(/\\/g, '').replace(/"\{/g, '{').replace(/\}"/g, '}');
+  }
+
+  async openWebSocket(uri: string): Promise<any> {
+    const userId = this.user.id;
+    const sessionId = this.sessionID;
+    const { IBMIntegrationId } = this;
+    return new Promise((resolve, reject) => {
+      try {
```

Comment:
This try...catch is unnecessary both because you should just allow this function to throw but also because there isn't really anything inside of it that can actually throw anything. The `new WebSocket` will only throw if there is a syntax error in the URL but otherwise there's nothing here to throw anything.